src/main/java/org/jcodec/containers/mp4/muxer/CodecMP4MuxerTrack.java

```diff
@@ -117,14 +118,11 @@ public void addFrameInternal(Packet pkt, int entryNo) throws IOException {
     protected Box finish(MovieHeaderBox mvhd) throws IOException {
         checkState(!finished, "The muxer track has finished muxing");
         if (getEntries().isEmpty()) {
-            if (codec == Codec.H264) {
+            if (codec == Codec.H264 && !spsList.isEmpty()) {
                 SeqParameterSet sps = SeqParameterSet.read(spsList.get(0).duplicate());
                 Size size = H264Utils.getPicSize(sps);
-                VideoCodecMeta meta = org.jcodec.common.VideoCodecMeta.createSimpleVideoCodecMeta(size,
-                        ColorSpace.YUV420);
+                VideoCodecMeta meta = createSimpleVideoCodecMeta(size, ColorSpace.YUV420);
                 addVideoSampleEntry(meta);
-            } else {
-                checkState(false, "Sample entry missing not supported for anything other then H.264");
```

Comment:
I would convert this to a warning, i.e. Logger.warn("Creating a track without sample entry");